src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/provisioner/src/support.py

```diff
@@ -0,0 +1,58 @@
+"""Support module used throughout the ADF
+"""
+
+from botocore.config import Config
+from botocore.exceptions import ClientError, BotoCoreError
+from logger import configure_logger
+
+from .account import Account
+
+
+LOGGER = configure_logger(__name__)
+
+class Support: # pylint: disable=R0904
+    """Class used for accessing AWS Support API
+    """
+
+    _config = Config(retries=dict(max_attempts=30))
+
+    def __init__(self, role):
+        self.client = role.client("support", region_name='us-east-1', config=Support._config)
+
+    def check_aws_support_subscription(self):
+        try:
+            self.client.describe_cases()
+            return True
+        except (ClientError, BotoCoreError) as e:
+            if e.response["Error"]["Code"] == "SubscriptionRequiredException":
+                LOGGER.warning("Failed subscription for account")
+                LOGGER.info('Enterprise Support is not enabled')
+                return False
+            raise e
+
+    def enable_enterprise_support_for_account(self, account: Account, account_id):
+        try:
+            cc_email = account.email
```

Comment:
Could you add a debug log message that states that we are about to create a ticket with its context?
That makes debugging easier in the future.